backend/routes/auth_route.py

```diff
@@ -91,4 +93,51 @@ def verify_password_hash(password, password_salt, stored_password_hash):
     """
     hashed_password = hashlib.sha256((password + password_salt).encode('utf-8')).hexdigest()
     
-    return hashed_password == stored_password_hash
\ No newline at end of file
+    return hashed_password == stored_password_hash
+
+@auth_bp.route('/enable_2fa', methods=['POST'])
+@jwt_required()
+@cross_origin()
+def enable_2fa():
+    user = Users.objects(username=current_user.username).first()
+    user.is_two_factor_auth_enabled = True
+    user.generate_totp_secret()
+    user.save()
+
+    uri = user.get_authentication_setup_uri()
+    base64_qr_image = get_b64encoded_qr_image(uri)
+
+    return jsonify({'qr_image': base64_qr_image})
+
+@auth_bp.route('/disable_2fa', methods=['POST'])
+@jwt_required()
+@cross_origin()
+def disable_2fa():
+    user = Users.objects(username=current_user.username).first()
+    user.is_two_factor_auth_enabled = False
+    del user.totp_secret
+    user.save()
+
+    return jsonify({'message': 'Two-factor authentication disabled successfully'})
+
+@auth_bp.route('/verify_otp', methods=['POST'])
+@jwt_required()
+@cross_origin()
+def verify_otp():
+    otp = request.json.get('otp')
+    if not otp:
+        return jsonify({'message': 'OTP is required'}), 400
+
+    user = Users.objects(username=current_user.username).first()
+    if not user:
+        return jsonify({'message': 'User not found'}), 404
+
+    if not user.is_two_factor_auth_enabled:
+        return jsonify({'message': 'Two-factor authentication is not enabled'}), 403
+    
+    if user.is_otp_valid(otp):
+        flash("2FA verification successful. You are logged in!", "success")
```

Comment:
In this reference I see the following text about "flash": "In other words, the flash() method of the flask module passes the message to the next request which is an HTML template. The syntax to use the flash() method is given below." [[ref]](https://www.javatpoint.com/flask-flashing) I am not sure if it is useful to use the flash method when we use REST-API, I could be wrong please correct me 🙏🏻